### PR TITLE
ARIA attributes: serialize empty Hash and Array as nil

### PR DESCRIPTION
**Pull request by seanpdoyle**
### Summary

As a follow-up to [rails/rails#40479][], ensure that empty Hash and
Array arguments are treated as `nil`.

For example, when conditionally rendering an [aria-describedby][]
attribute to associate an input with a related validation error, treat
an empty Array as `nil`, an omit the attribute entirely:

```html+erb
<% post = Post.new %>

<%= form_with model: post do |form| %>
  <%= form.text_field :title, aria: { describedby: { post_title_error: post.errors[:title].any? } } %>
  <%= tag.span(post.errors[:title].to_sentence, id: :post_title_error) if post.errors[:title].any? %>
<% end %>
```

In this example, when there are no errors, the desired outcome is for
the `<input type="text" name="post[title]">` element to _omit_ the
`[aria-describedby="post_title_error"]` attribute, and to only include
it when there are errors on the `title` attribute.

Without this change, the desired outcome can be achieved with a
combination of a `#token_list` and `#presence` call:

```diff
<% post = Post.new %>

<%= form_with model: post do |form| %>
-  <%= form.text_field :title, aria: { describedby: {post_title_error: post.errors[:title].any?} }
+  <%= form.text_field :title, aria: { describedby: token_list(post_title_error: post.errors[:title].any?).presence } %>
  <%= tag.span(post.errors[:title].to_sentence, id: :post_title_error) if post.errors[:title].any? %>
<% end %>
```

[rails/rails#40479]: https://github.com/rails/rails/pull/40479
[aria-describedby]: https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA21#example-2-identifying-errors-in-data-format

### CODE REVIEW

File: actionview/test/template/tag_helper_test.rb [hunk 1]
```diff
@@ -449,11 +449,11 @@ def test_data_attributes
   def test_aria_attributes
     ["aria", :aria].each { |aria|
       assert_dom_equal '<a aria-a-float="3.14" aria-a-big-decimal="-123.456" aria-a-number="1" aria-truthy="true" aria-falsey="false" aria-array="1 2 3" aria-hash="a b" aria-tokens="a b" aria-string-with-quotes="double&quot;quote&quot;party&quot;" aria-string="hello" aria-symbol="foo" />',
-        tag("a", aria => { nil: nil, a_float: 3.14, a_big_decimal: BigDecimal("-123.456"), a_number: 1, truthy: true, falsey: false, string: "hello", symbol: :foo, array: [1, 2, 3], hash: { a: true, b: "truthy", falsey: false, nil: nil }, tokens: ["a", { b: true, c: false }], string_with_quotes: 'double"quote"party"' })
+        tag("a", aria => { nil: nil, a_float: 3.14, a_big_decimal: BigDecimal("-123.456"), a_number: 1, truthy: true, falsey: false, string: "hello", symbol: :foo, array: [1, 2, 3], empty_array:[], hash: { a: true, b: "truthy", falsey: false, nil: nil }, empty_hash: {}, tokens: ["a", { b: true, c: false }], empty_tokens: [{ a: false }], string_with_quotes: 'double"quote"party"' })
```
Comment (seanpdoyle) [inline actionview/test/template/tag_helper_test.rb:452]: @rafaelfranca @kaspth I realized that the original PR (#40479) missed this edge case, and serializes `[]` and `{}` as `""`. I've added coverage here. What do you think?

File: actionview/lib/action_view/helpers/tag_helper.rb [hunk 2]
```diff
@@ -84,7 +84,15 @@ def tag_options(options, escape = true)
               value.each_pair do |k, v|
                 next if v.nil?
 
-                v = (v.is_a?(Array) || v.is_a?(Hash)) ? safe_join(TagHelper.build_tag_values(v), " ") : v.to_s
+                case v
+                when Array, Hash
+                  tokens = TagHelper.build_tag_values(v)
+                  next if tokens.none?
```
Comment (seanpdoyle) [inline actionview/lib/action_view/helpers/tag_helper.rb:90]: Without this change, the desired outcome can be achieved with a combination of a `#token_list` and `#presence` call:

```html+erb
<% post = Post.new %>

<%= form_with model: post do |form| %>
  <%= form.text_field :title, aria: { describedby: token_list(post_title_error: post.errors[:title].any?).presence } %>
  <%= tag.span(post.errors[:title].to_sentence, id: :post_title_error) if post.errors[:title].any? %>
<% end %>
```

File: actionview/test/template/tag_helper_test.rb (hunk 1 above)
Comment (seanpdoyle) [reply actionview/test/template/tag_helper_test.rb]: Another alternative as a caller if this doesn't land is to wrap the value in an inline conditional: 

```rb
aria: { describedby: (:key if truthy) }
```

Though when part of a multi-element array, this could get messy.

**Comment by kaspth** [comment]
Perfect! In cases where we can infer what the proper outcome is, we should, most often, do it 😄
